taskpools/primitives/barriers_posix.nim

```diff
@@ -31,15 +35,15 @@ when defined(osx):
   export pthread_barrier_init, pthread_barrier_wait, pthread_barrier_destroy
 else:
   proc pthread_barrier_init*(
-        barrier: PthreadBarrier,
-        attr: PthreadAttr or ptr PthreadAttr,
+        barrier: var PthreadBarrier,
+        attr: ptr PthreadAttr,
```

Comment:
with `{.byref.}` `ptr` is actually not needed.

Reply:
yeah, this is a bit of a matter of style - either `byref` is used, or `ptr`/`var` in the function declaration - since `c2nim` prefers the latter, it makes sense to move in that direction, in general .. though it doesn't interact well with `sink` it seems :/ need to think a bit more about this

Reply:
`var` is orthogonal to byref/byval/nimcall, it should be used for mutation. A `byref` type will not have double pointer if we use `var MyType`

Reply:
#21 demonstrates the difference - for example, the original `pthread_barrier_init` was _not_ `var`, even though clearly it modifies the parameter - also, the macos `destroy` was not `sink` - that said, it doesn't _really_ matter that much, both solve the problem even if the current posix definitions are a bit random